ports/boost-asio/b2-options.cmake

```
@@ -0,0 +1,19 @@
+if(CMAKE_BUILD_TYPE STREQUAL "Release")
+    set(lib_path_suffix lib)
+else()
+    set(lib_path_suffix debug/lib)
+endif()
+
+if("ssl" IN_LIST FEATURES)
+  function(isolate_asio_with_ssl)
+    find_library(BOOST_ASIO_OPENSSL_CRYPTO_LIBRARY_PATH NAMES crypto libcrypto NAMES_PER_DIR HINTS "${CURRENT_INSTALLED_DIR}/${lib_path_suffix}" NO_CACHE REQUIRED)
+    find_library(BOOST_ASIO_OPENSSL_SSL_LIBRARY_PATH NAMES ssl libssl NAMES_PER_DIR HINTS "${CURRENT_INSTALLED_DIR}/${lib_path_suffix}" NO_CACHE REQUIRED)
+    list(APPEND B2_OPTIONS
+      asio-with-ssl=yes
+      -sASIO_CRYPTO_LIBRARY_PATH="${BOOST_ASIO_OPENSSL_CRYPTO_LIBRARY_PATH}"
+      -sASIO_SSL_LIBRARY_PATH="${BOOST_ASIO_OPENSSL_SSL_LIBRARY_PATH}"
+    )
+    set(B2_OPTIONS ${B2_OPTIONS} PARENT_SCOPE)
+  endfunction()
+  isolate_asio_with_ssl()

```

```suggestion
  find_library(BOOST_ASIO_OPENSSL_CRYPTO_LIBRARY_PATH NAMES crypto libcrypto NAMES_PER_DIR HINTS "${CURRENT_INSTALLED_DIR}/${lib_path_suffix}" NO_CACHE REQUIRED)
  find_library(BOOST_ASIO_OPENSSL_SSL_LIBRARY_PATH NAMES ssl libssl NAMES_PER_DIR HINTS "${CURRENT_INSTALLED_DIR}/${lib_path_suffix}" NO_CACHE REQUIRED)
  list(APPEND B2_OPTIONS
    asio-with-ssl=yes
   -sASIO_CRYPTO_LIBRARY_PATH="${BOOST_ASIO_OPENSSL_CRYPTO_LIBRARY_PATH}"
    -sASIO_SSL_LIBRARY_PATH="${BOOST_ASIO_OPENSSL_SSL_LIBRARY_PATH}"
  )
```

I'm not seeing an obvious reason to put this into an immediately invoked function?

---

The "isolated" in the function name is a hint. It stops many if not most cmake variables set by any cmake scripts invoked by `find_library` from leaking outside the function where they may affect other cmake. It may be overkill, but I have learned from bitter experience always isolate all potential external cmake script calls in cmake by default, and if you ended up typing unnecessary cmake, that's better than losing several work days to tracking down spurious cmake misoperation.

---

(This is super nitpick level, feel free to ignore)